sonar-plugin-api/src/main/java/org/sonar/api/server/rule/internal/DefaultNewRule.java

```diff
@@ -134,13 +142,45 @@ public DefaultNewRule setType(RuleType t) {
 
   @Override
   public RulesDefinition.NewRule addDescriptionSection(RuleDescriptionSection ruleDescriptionSection) {
-    checkArgument(isSectionKeyUnique(ruleDescriptionSection.getKey()), "A section with key %s already exists", ruleDescriptionSection.getKey());
+    assertRuleDescriptionSectionIsValid(ruleDescriptionSection);
     ruleDescriptionSections.add(ruleDescriptionSection);
     return this;
   }
 
+  private void assertRuleDescriptionSectionIsValid(RuleDescriptionSection ruleDescriptionSection) {
```

Comment:
I actually tried this way while implementing, but thought the second method argument as an else made it slightly less readable. But I'd be happy to hear feedback in case someone has an opinion here?

Reply:
I will leave it as it is as I personally think it's more readable.